test/fib-dbg.wasm.fromBinary

```diff
@@ -133,6 +133,7 @@
      (i32.const 0)
     )
    )
+   ;;@ fib.c:3:0
```

Comment:
@kripken Hi Alon,
`sourceMap` location is not one by one equal to the debug location of expression. it is a range concept.
to be short, if expression offset is between previous source map debug location offset and current source map debug location offset, this expression will take the previous source map debug location.
before this change, the debug location of siblings element just were not printed, but it existed in the `functionRef->debugLocations`, so this change just print it and it is more clear to see which instruction is under which location.

Reply:
if remove the filter condition of `printDebugLocation` of `Print.cpp`.
it will print all locations to all expressions, and it is more clear to see the location information, and just for debug, print all debug location is not neccessary I suppose.

Reply:
Is the range concept idea documented somewhere? I'm not familiar with that. I had assumed source maps are just individual locations, but I could be wrong...

Reply:
Hi Alon,
I didnot find the documentation of the solution, this may come from PR:
https://github.com/WebAssembly/binaryen/pull/1017
Here is the read and write flow:
1. `wasm-opt` read:
`wasm-s-parser.h` function: `SExpressionParser::parseDebugLocation` it will update `loc` of `SExpressionParser`, so the after expression will take this position until the `loc` will updated with another debug location.
2. binary write:
`WasmBinaryWriter::writeSourceMapEpilog` 
```
writeBase64VLQ(*sourceMap, int32_t(offset - lastOffset));
writeBase64VLQ(*sourceMap, int32_t(loc->fileIndex - lastLoc.fileIndex));
writeBase64VLQ(*sourceMap, int32_t(loc->lineNumber - lastLoc.lineNumber));
writeBase64VLQ(*sourceMap,
                 int32_t(loc->columnNumber - lastLoc.columnNumber));
```

Reply:
@kripken 
Hi Alon,
Do you think source map should be individual point? does it cause impact to wabt or others tool?
Regards

Reply:
@kripken 
Hi Alon,
can you give some advice based on this? thank you.
Regards

Reply:
@kripken 
Hi Alon,
I suppose the primary design for source map is shrink the size of debug info like DWARF, but source map is lack the control instructions, so do you think we should back to `point to point` solution for source map debug info?

Reply:
@kripken 
Hi Alon,
the changes contains two parts:
part 1: during printing, if parent ancestors node already printed the debug info, skip print
part 2: during source map generation, only ancestors generate the source map debug info
I am not sure if it is clear, if you have any question, please feel free to ask me

Reply:
I think I understand that, and the code sort of makes sense. But I am having specific trouble in reading the diffs in the testcase specifically. That is where I need help. I'll add a comment on a specific part of the test diff now.

Reply:
Oh, sorry, this is already on a test diff :smile: Then, yeah: how can I see that this addition on line 136 is a correct one to add? I'm not sure how to read the sourcemap itself and see that it says there should be an annotation here.